packages/flutter_tools/bin/xcode_backend.dart

```diff
@@ -32,41 +32,79 @@ class Context {
   final List<String> arguments;
   RandomAccessFile? scriptOutputStream;
 
+  static const String incompatibleErrorMessage =
+      'Your Xcode project is incompatible with this version of Flutter. '
+      'Run "rm -rf ios/Runner.xcodeproj" and "flutter create ." to regenerate.\n';
+
   void run() {
     if (arguments.isEmpty) {
       // Named entry points were introduced in Flutter v0.0.7.
-      stderr.write(
-        'error: Your Xcode project is incompatible with this version of Flutter. '
-        'Run "rm -rf ios/Runner.xcodeproj" and "flutter create ." to regenerate.\n',
-      );
+      echoXcodeError(incompatibleErrorMessage);
       exit(-1);
     }
 
-    final String subCommand = arguments.first;
+    final String? subCommand = validateCommand(arguments[0]);
+    if (subCommand == null) {
+      echoXcodeError(incompatibleErrorMessage);
+      exit(-1);
+    }
+    final String? platformName = arguments.length < 2 ? null : arguments[1];
+    final TargetPlatform platform = parsePlatform(platformName);
     switch (subCommand) {
       case 'build':
-        buildApp();
+        buildApp(platform);
       case 'prepare':
-        prepare();
+        prepare(platform);
       case 'thin':
         // No-op, thinning is handled during the bundle asset assemble build target.
         break;
       case 'embed':
-        embedFlutterFrameworks();
       case 'embed_and_thin':
         // Thinning is handled during the bundle asset assemble build target, so just embed.
-        embedFlutterFrameworks();
+        embedFlutterFrameworks(platform);
       case 'test_vm_service_bonjour_service':
         // Exposed for integration testing only.
         addVmServiceBonjourService();
     }
   }
 
+  /// Validates the command argument matches one of the possible commands.
+  /// Returns null if not.
+  String? validateCommand(String command) {
```

Comment:
Could we make this method and `parseFlutterBuildMode` mirror each other? Currently, `validateCommand` returns `null` on error and expects its caller print an error + exit. However, `parseFlutterBuildMode` prints its error and exits. I don't have a preference on which approach to take, but ideally these two methods would take the same approach :)